.github/workflows/test.yml

```diff
@@ -0,0 +1,51 @@
+name: test
+
+on:
+  workflow_call:
+    inputs:
+      php:
+        default: "8.3"
+        type: string
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.php }}
+          tools: composer:v2
+          coverage: none
+
+      - name: Install Composer
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer install --no-interaction --no-progress
+
+      - name: Install NPM
+        run: npm ci
+
+      - name: Configure Environment
+        run: |
+          cp .env.example .env
+          php artisan key:generate
+
+      - name: Prepare Database
+        run: |
+          php artisan migrate --force
```

Comment:
Note the `--force` flag is used rather than touching the sqlite db file, as that should produce the same output via Laravel's latest migrate command.

Although we can change it back if needed.